libraries/chain/block_header_state.cpp

```diff
@@ -197,40 +202,21 @@ block_header_state block_header_state::next(const signed_block_header& h, valida
    auto  if_entry = exts.lower_bound(instant_finality_extension::extension_id());
    auto& if_ext   = std::get<instant_finality_extension>(if_entry->second);
 
-   building_block_input bb_input{
-      .parent_id        = block_id,
-      .parent_timestamp = timestamp(),
-      .timestamp        = h.timestamp,
-      .producer         = producer,
-      .new_protocol_feature_activations = std::move(new_protocol_feature_activations)
-   };
-
-   digest_type action_mroot = {};
-
    if (h.is_proper_svnn_block()) {
       // if there is no Finality Tree Root associated with the block,
       // then this needs to validate that h.action_mroot is the empty digest
       auto next_core_metadata = core.next_metadata(if_ext.qc_claim);
       bool no_finality_tree_associated = core.is_genesis_block_num(next_core_metadata.final_on_strong_qc_block_num);
 
-      EOS_ASSERT( no_finality_tree_associated == h.action_mroot.empty(),
-         block_validate_exception,
-         "No Finality Tree Root associated with the block test does not match with empty action_mroot test: no finality tree associated (${n}), action_mroot empty (${e}), final_on_strong_qc_block_num (${f})",
-         ("n", no_finality_tree_associated)("e", h.action_mroot.empty())("f", next_core_metadata.final_on_strong_qc_block_num));
-
-      action_mroot = h.action_mroot;
```

Comment:
Where do you set `action_mroot` now? It is needed in the header (the finality root).

Reply:
The entire header is copied above.